src/java.base/share/classes/jdk/internal/module/ModuleInfoWriter.java

```diff
@@ -26,17 +26,19 @@
 
 import java.io.IOException;
 import java.io.OutputStream;
+import java.lang.constant.ClassDesc;
+import java.lang.module.InvalidModuleDescriptorException;
 import java.lang.module.ModuleDescriptor;
 import java.nio.ByteBuffer;
 import java.util.Map;
-import java.util.stream.Stream;
-
-import jdk.internal.org.objectweb.asm.ClassWriter;
-import jdk.internal.org.objectweb.asm.ModuleVisitor;
-import jdk.internal.org.objectweb.asm.Opcodes;
-import jdk.internal.org.objectweb.asm.commons.ModuleResolutionAttribute;
-import jdk.internal.org.objectweb.asm.commons.ModuleTargetAttribute;
-import static jdk.internal.org.objectweb.asm.Opcodes.*;
+import java.util.Set;
+import jdk.internal.classfile.Classfile;
+import jdk.internal.classfile.jdktypes.ModuleDesc;
+import jdk.internal.classfile.jdktypes.PackageDesc;
```

Comment:
I'm a bit surprised of `jdk.internal.classfile.jdktypes` package.   ClassFile API is JDK-specific.  Would `jdk.internal.classfile.constant` be more appropriate?